### ISSUE TITLE
polytope bug

### ISSUE THREAD

**Issue by scungao** [dreal/dreal3#246]
The polytope option has a bug at the boundary of the domain of some functions. For instance: 

```
(set-logic QF_NRA)
(declare-fun x1 () Real)
(declare-fun x2 () Real)
(assert (<= -10.0 x1))
(assert (<= x1 10.0))
(assert (<= -10.0 x2))
(assert (<= x2 10.0))
(assert (= x1 1))
(assert (= x2 (asin x1)))
(check-sat)
(exit)
```

It checks whether x2 = arcsin(x1) has a solution at x1=1. With polytope option it returns unsat (without it the answer is correct). Same behavior for arccos.

**Comment by soonhokong** [comment]
I checked that this bug is not from IBEX. 

There is a bug in constructing ibex system from dReal side. I'm fixing it.

**Comment by soonhokong** [comment]
The given system is square (two variables and two equalities -- `x1 = 1` and `x2 = asin(x1)`). For a square system, we call newton contractor in addition to polytope hull contractor. It seems that the problem comes from newton. I'll check IBEX part first..

**Comment by soonhokong** [comment]
For example, if we replace:

```
(assert (= x1 1))
```

with

```
(assert (<= x1 1))
(assert (>= x1 1))
```

the problem goes away. Of course, the problem is still there.

**Comment by soonhokong** [comment]
I can confirm that this is IBEX issue. I'll open an issue at upstream.

**Comment by soonhokong** [comment]
Here is a repro-step:

```c++
#include <algorithm>
#include <iostream>
#include "ibex/ibex.h"

using ibex::Variable;
using ibex::Function;
using ibex::NumConstraint;
using ibex::CtcFwdBwd;
using ibex::IntervalVector;
using ibex::Interval;
using ibex::ExprSymbol;
using std::cout;
using std::endl;

int main() {
    Variable x1,x2;
    Function f(x1,x2,ibex::Return(x1 - 1, ibex::asin(x2) - x1));
    double init_box[][2]={{1, 1}, {1.570796326794893, 1.570796326794901}};
    IntervalVector box(2,init_box);
    ibex::CtcNewton newton(f);
    cout << "Before: " << box << endl;
    newton.contract(box);
    cout << "After: " << box << endl;
    return 0;
}
```

You have:
```
Assertion failed: (!v.is_empty()), function _mid, file /Users/soonhok/work/dreal3/build/debug-clang/external/src/IBEX/src/arithmetic/ibex_TemplateVector.h_, line 116.
[1]    91254 abort      ./tests/ibex/ibex_newton_01
```

**Comment by soonhokong** [comment]
An issue is created at IBEX repo. I'll update it when I get a response from the team.

**Comment by soonhokong** [comment]
The upstream bug is fixed by the ibex team and I've updated `dreal-deps/ibex` repo. Please do `make update_ibex && make`. It will do the update and rebuild dReal. I'll release a new version soon (for this one + CAPD power bug reproted by Nicola).

**Comment by scungao** [comment]
nice:)

**Comment by soonhokong** [comment]
Fixed by https://github.com/dreal-deps/ibex-lib/commit/59642a0a5e29d0f1d5f772894904291c4974b2d5. Users need to do `make update_ibex`.

**Comment by soonhokong** [comment]
> Fixed by dreal-deps/ibex-lib@59642a0.

IBEX team confirmed this patch.
